Gems/ScriptCanvas/Code/Include/ScriptCanvas/Grammar/ParsingUtilitiesScriptEventExtension.cpp

```diff
@@ -0,0 +1,244 @@
+        return result;
+    }
+
+    FunctionNodeToScriptEventResult ParseScriptEvent(const Node& node)
+    {
+        FunctionNodeToScriptEventResult result{ false, &node, {}, {} };
+
+        auto functionDefinitionNode = azrtti_cast<const Nodes::Core::FunctionDefinitionNode*>(&node);
+
+        if (!functionDefinitionNode)
+        {
+            return result;
+        }
+
+        if (!functionDefinitionNode->IsExecutionEntry())
+        {
+            return result;
+        }
+
+        auto outcome = TranslateToScriptEventMethod(*functionDefinitionNode);
+        if (outcome.IsSuccess())
+        {
+            result.m_method = outcome.GetValue();
+        }
+        else
+        {
+            result.m_parseErrors.push_back(outcome.GetError());
+        }
+
+        result.m_isScriptEvent = result.m_parseErrors.empty();
+        return result;
+    }
+
+    GraphToScriptEventsResult ParseScriptEventsDefinition(const Graph& graph)
+    {
+        // get name, tool tip, category from variables
+        GraphToScriptEventsResult result = ParseMinimumScriptEventArtifacts(graph);
+        
+        {
+            auto nodes = graph.GetNodesOfType<Node>();
+            bool anyNodeWasInvalid = false;
+
+            for (auto node : nodes)
+            {
+                result.m_nodeResults.push_back(ParseScriptEvent(*node));
+                anyNodeWasInvalid = anyNodeWasInvalid || !result.m_nodeResults.back().m_parseErrors.empty();
+            }
+
+            if (anyNodeWasInvalid)
+            {
+                result.m_parseErrors.push_back("At least one node failed to parse as an event.");
+            }
+        }
+
+        // get address type from variables
+        auto variableData = graph.GetVariableDataConst();
+        if (!variableData)
+        {
+            result.m_parseErrors.push_back("Missing variable data in graph. Parsing can not continue.");
+            return result;
+        }
+
+        for (auto& nodeResult : result.m_nodeResults)
+        {
+            if (nodeResult.m_isScriptEvent)
+            {
+                result.m_event.NewMethod() = nodeResult.m_method;
+            }
+        }
+
+        if (auto eventOutcome = result.m_event.Validate(); !eventOutcome.IsSuccess())
+        {
+            result.m_parseErrors.push_back(eventOutcome.GetError());
+        }
+
+        AZ::IO::ByteContainerStream stream(&result.m_event.ModScriptCanvasSerializationData());
+        auto graphSerializationResult = ScriptCanvas::Serialize(*graph.GetOwnership(), stream);
+
+        if (!graphSerializationResult.m_isSuccessful)
+        {
+            result.m_parseErrors.push_back(graphSerializationResult.m_errors);
+        }
+
+        result.m_isScriptEvents = result.m_parseErrors.empty();
+        return result;
+    }
+}
+
+namespace ScriptEvents
+{
+    void AddScriptEventHelpers(ScriptCanvas::Graph& graph)
+    {
+        using namespace ScriptCanvas;
+
```

Comment:
This particular use case of using namespace ScriptCanvas is actually OK, because it is restricted to function scope.